src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+            ...
+            LookupError: vertex (Michael) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
+        return self._slice(i)
+
+    def xslice_sequence(self, v):
+        r"""
+        Return the x-slice sequence of the vertex `v`.
+
+        INPUT:
+
+        - ``v`` -- a vertex of the graph corresponding to the slice
+          decomposition.
+
+        OUTPUT:
+
+        A list of list corresponding to the x-slice sequence of ``v``.
+
+        EXAMPLES:
+
+        ::
+
+            sage: G = Graph('L~mpn~Nrv{^o~_').relabel('abcdefguvwxyz',inplace=False)
+            sage: SD = G.slice_decomposition(initial_vertex='x')
+            sage: SD.xslice_sequence('x')
+            [['x'], ['a', 'b', 'c', 'd', 'e', 'f', 'g'], ['u', 'y', 'z'], ['v', 'w']]
+            sage: SD.xslice_sequence('a')
+            [['a'], ['b', 'c', 'd', 'e', 'f'], ['g']]
+
+        The flatten x-slice sequence of a vertex corresponds to the slice of the
+        same vertex::
+
+            sage: from itertools import chain
+            sage: all(list(chain(*SD.xslice_sequence(v))) == SD.slice(v) \
+            ....:       for v in G)
+            True
+
+        The first list of the sequence is always a singleton containing the
+        input vertex::
+
+            sage: all(SD.xslice_sequence(v)[0] == [v] for v in G)
+            True
+
+        If the length of the slice if more than 1, the second list of the
+        sequence is either, all the remaining vertices of the slice of `v`, if
+        `v` is isolated in the subgraph induced by the slice of `v`, or the
+        neighbors of `v` in the subgraph induced by the slice of `v`::
+
+            sage: all(SD.xslice_sequence(v)[1] == SD.slice(v)[1:] for v in G \
+            ....:           if G.subgraph(SD.slice(v)).degree(v) == 0 \
+            ....:               and len(SD.slice(v)) > 1)
+            True
+            sage: for v in G:
+            ....:     if len(SD.slice(v)) > 1:
+            ....:         xslice_seq = SD.xslice_sequence(v)
+            ....:         S = G.subgraph(SD.slice(v))
+            ....:         if S.degree(v) > 0:
+            ....:             set(xslice_seq[1]) == set(S.neighbor_iterator(v))
+            True
+            True
+            True
+            True
+
+        TESTS:
+
+            sage: SD = graphs.PetersenGraph().slice_decomposition()
+            sage: SD.xslice_sequence('Terry')
+            Traceback (most recent call last):
+            ...
+            LookupError: vertex (Terry) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
```

Comment:
`cdef int i = ...` ?

Reply:
done in f3d502e0a17cc39325d09f1d4a00c60dd3aa278e